sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala

```diff
@@ -1095,16 +1095,13 @@ class GpuMergeAggregateIterator(
           closeOnExcept(new ArrayBuffer[AutoClosableArrayBuffer[SpillableColumnarBatch]]) {
             toAggregateBuckets =>
               var currentSize = 0L
-              while (batchesByBucket.nonEmpty &&
-                (
-                  // for some test cases targetMergeBatchSize is too small to fit any bucket,
-                  // in this case we put the first bucket into toAggregateBuckets anyway
-                  // refer to https://github.com/NVIDIA/spark-rapids/issues/11790 for examples
-                  toAggregateBuckets.isEmpty ||
-                    batchesByBucket.last.size() + currentSize <= targetMergeBatchSize)) {
-                val bucket = batchesByBucket.remove(batchesByBucket.size - 1)
-                currentSize += bucket.map(_.sizeInBytes).sum
-                toAggregateBuckets += bucket
+              var keepGoing = true
+              while (batchesByBucket.nonEmpty && keepGoing) {
```

Comment:
may need use a separate PR for this as it is irrelevant to the description in https://github.com/NVIDIA/spark-rapids/issues/11903

Reply:
Thx for review. @binmahone Could you help file an issue for this? Then I will follow your suggestion.

Reply:
you can also consider modifying  the description in 11903 :-)

Reply:
Updated